service/controller/v3/resource/vnetpeering/update.go

```diff
@@ -0,0 +1,133 @@
+package vnetpeering
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2017-09-01/network"
+	providerv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/controller"
+
+	"github.com/giantswarm/azure-operator/service/controller/v3/key"
+)
+
+// NewUpdatePatch provide a controller.Patch holding the needed network.VirtualNetworkPeering update to have current comply with desired.
+func (r *Resource) NewUpdatePatch(ctx context.Context, azureConfig, current, desired interface{}) (*controller.Patch, error) {
```

Comment:
I am lost now. I thought this here is just to delete the orphaned peering. Why do we need this update magic now?